crates/ty_python_semantic/src/types/ide_support.rs

```diff
@@ -232,16 +247,59 @@ impl AllMembers {
             let index = semantic_index(db, file);
             for function_scope_id in attribute_scopes(db, class_body_scope) {
                 let place_table = index.place_table(function_scope_id);
-                self.members
-                    .extend(place_table.instance_attributes().cloned());
+                for place_expr in place_table.places() {
+                    let Some(name) = place_expr.as_instance_attribute() else {
+                        continue;
+                    };
+                    let result = class_literal.own_instance_member(db, name.as_str());
+                    let Some(ty) = result.place.ignore_possibly_unbound() else {
+                        continue;
+                    };
+                    self.members.insert(Member {
+                        name: name.clone(),
+                        ty,
+                    });
+                }
             }
         }
     }
 }
 
+#[derive(Clone, Debug)]
+pub struct Member<'db> {
+    pub name: Name,
+    pub ty: Type<'db>,
+}
```

Comment:
Can you document why the equality only takes the name into account but not its type